sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Usage:
+#   ExternalZephyrProject_Add(...)
+#
+# This function includes a Zephyr based build system into the multiimage
+# build system
+#
+# APPLICATION: <name>: Name of the application, name will also be used for build
+#                      folder of the application
+# SOURCE_DIR <dir>:    Source directory of the application
+# BOARD <board>:       Use <board> for application build instead user defined BOARD.
+# MAIN_APP:            Flag indicating this application is the main application
+#                      and where user defined settings should be passed on as-is
+#                      except for multi image build flags.
+#                      For example, -DCONF_FILES=<files> will be passed on to the
+#                      MAIN_APP unmodified.
+#
+function(ExternalZephyrProject_Add)
+  cmake_parse_arguments(ZBUILD "MAIN_APP" "APPLICATION;BOARD;SOURCE_DIR" "" ${ARGN})
+
+  # TODO: Argument verification. Not required for the prototype but important later.
+  #       If this comment is still present when PR is not in draft state, please make a review comment here.
```

Comment:
Argument verification comment is still present

Reply:
I guess I had a sixth sense here when making the initial code :laughing:

Reply:
done.

Only unparsed args needs to be handled for now as invalid source dir / duplicated application name is taken care of by CMake internal target handling.